dap-mode.el

```diff
@@ -905,6 +968,45 @@ PARAMS are the event params.")
          (run-hook-with-args 'dap-loaded-sources-changed-hook debug-session)))
       (_ (dap-handle-event (intern event-type) debug-session body)))))
 
+(defcustom dap-default-terminal-kind "integrated"
+  "Default terminal type used for :console."
+  :type '(radio
+          (const :tag "Terminal within Emacs" :value "integrated")
+          (const :tag "External terminal program (`dap-external-terminal')"
+                 :value "external"))
+  :group 'dap-mode)
+
+(defun dap--start-process (debug-session parsed-msg)
+  (-let* (((&hash "arguments" (&hash? "args" "cwd" "title" "kind") "seq")
+           parsed-msg)
+          (default-directory cwd)
+          (command-to-run (string-join args " ")))
+    (pcase (or kind dap-default-terminal-kind)
+      ("external"
```

Comment:
external wont work unless xterm is defined. I suggest checking if xterm is present and if it is not present - show warning with instruction and use internal terminal. This is crucial for OOTB experience.

Reply:
Fixed. The entire block is wrapped in with-demoted-errors and will fall-back to using the built-in one instead. I tested with `dap-external-terminal' set to '("foobar") and with it set to my default KiTTY list, both working as expected. However, here I would like some more feedback: do you think with-demoted-errors is enough, or should I use condition-case + lsp--warn instead? Also, what do you think of the error message:
         (with-demoted-errors "dap-debug: failed to start \
external terminal: %s. Set `dap-external-terminal' to the correct
value or install the terminal configured (probably xterm)." ...)

Reply:
I merged it as it is but I am a bit concerned here because there might be different errors coming from this block and the error message will be misleading.

Reply:
Well, all errors would be related to failure to start the terminal, which could be for a variety of reasons: terminal not installed/not in PATH, terminal *is* installed but permission is denied to launch it (SELinux, -x bit), which would handle all of that. Is your worry that dap--send-message might fail? I'll fix that. Aside from that, what do you propose?

Reply:
I think it is good enough for now, lets see how it goes. A minor improvement would be to include the actual command(after replacement) that has failed in the error message.

Reply:
Would that even be possible using with-demoted-errors? The format string has to be static, doesn't it? Try evaluating the following in scratch:
```emacs-lisp
(let ((m "FooBAr%S"))
  (with-demoted-errors m
    (error "5")))
```
An error is still shown and there is no trace of m (as an error format) in macrostep.

Reply:
The fix to move dap--send-message outside of with-demoted-errors is in master.

Reply:
> Would that even be possible using with-demoted-errors? 

with-demoted-errors is just a wrapper over `condition-case-unless-debug`

Reply:
Fixed in master. What I was afraid of is that the code could become horrible ugly (in terms of indent). Long string constants never look elegant....